### DISCUSSION TITLE
Excluding paths doesn't work without includes

### DISCUSSION THREAD

**Discussion by yelizariev** [community/community#25369] [Actions]
<p>If I have in <code>paths</code>  excludes only, it will not work:</p>
<pre><code class="lang-auto">on:
  push:
    paths:
    - '!sub-project/docs/**'
</code></pre>
<p>What I suppose is to trigger workflow on all changes but <code>sub-project/docs/</code>. What I get is triggering on any changes</p>
<p>It either should be fixed, or some doc updates is required.</p>
<p>If it’s fixed, then <code>paths-ignore</code> option would not needed IMO.</p>
<p>As for docs, some updates could be done here: <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#excluding-paths" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#excluding-paths</a><br>
It currently has:</p>
<pre><code class="lang-auto"># Excluding paths

You can exclude paths using two types of filters. You cannot use both of these filters for the same event in a workflow.

** paths-ignore - Use the paths-ignore filter when you only need to exclude path names.
** paths - Use the paths filter when you need to filter paths for positive matches and exclude paths.

</code></pre>
<p>It should be stressed somehow that using excluds only in <code>paths</code> will not work</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I agree that the documentation should be more clear. In practice you can work around the limitation by having a blanket include and then exclude the parts you want:</p>
<pre><code class="lang-yaml">on:
  push:
    paths:
      - '**'
      - '!sub-project/docs/**'
</code></pre>

**Comment by yelizariev** [comment]
<p>That’s exactly what I did. Somehow I didn’t try <code>paths-ignore</code> when I found out the problem and end up with using `- ‘*/**’``</p>

**Comment by Yanjingzhu** [comment]
<p>I have created an internal ticket asking document team to update the document . Thank you for pointing this out and it is helpful to make a better GitHub Actions.</p>
